lptree.pas

```diff
@@ -4174,6 +4186,104 @@ constructor TLapeTree_InternalMethod_Raise.Create(ACompiler: TLapeCompilerBase;
   FForceParam := True;
 end;
 
+function TLapeTree_InternalMethod_Find.resType: TLapeType;
+begin
+  if (FResType = nil) then
+    FResType := FCompiler.getBaseType(ltInt32);
+
+  Result := inherited;
+end;
+
+function TLapeTree_InternalMethod_Find.Compile(var Offset: Integer): TResVar;
+var
+  _Find: TLapeGlobalVar;
+  Item, Arr: TResVar;
+begin
+  Result := NullResVar;
+  Dest := NullResVar;
+
+  if (FParams.Count <> 2) or isEmpty(FParams[0]) or isEmpty(FParams[1]) then
+    LapeExceptionFmt(lpeWrongNumberParams, [2], DocPos);
+
+  Item := FParams[0].Compile(Offset);
+  Arr := FParams[1].Compile(Offset);
+
+  if (Item.VarType.BaseType in LapeArrayTypes - LapeStringTypes) then // Lape does not have array comparison
+    LapeException(lpeInvalidEvaluation, DocPos);
+  if not (Arr.VarType.BaseType in LapeArrayTypes) then
+    LapeException(lpeInvalidEvaluation, DocPos);
+
+  _Find := FCompiler['_Find'];
+  Assert((_Find <> nil) and (_Find.VarType is TLapeType_OverloadedMethod));
+  _Find := TLapeType_OverloadedMethod(_Find.VarType).getMethod(getTypeArray([Item.VarType, Arr.VarType]), FCompiler.getBaseType(ltInt32));
+
+  with TLapeTree_Invoke.Create(_Find, Self) do
+  try
+    addParam(TLapeTree_ResVar.Create(Item.IncLock(), Self));
+    addParam(TLapeTree_ResVar.Create(Arr.IncLock(), Self));
+
+    Result := Compile(Offset);
+  finally
+    Item.Spill(1);
+    Arr.Spill(1);
+
+    Free();
+  end;
+end;
+
+function TLapeTree_InternalMethod_FindAll.resType: TLapeType;
+var
+  Decls: TLapeDeclArray;
+begin
+  if (FResType = nil) then
+  begin
+    Decls := FCompiler.ManagedDeclarations.getByClassAndName('!TIntegerArray', TLapeType_DynArray, bFalse);
```

Comment:
Not really a fan of doing this here. Can be moved to the other "!" type initializations, or just be created every time.